test/pubspec_test.dart

```diff
@@ -420,19 +442,6 @@ dependencies:
               "The 'name' key must have a string value without a minimum Dart "
                   'SDK constraint of 2.15.');
         });
-
-        test('and a direct url', () {
-          expectPubspecException(
-              '''
-name: pkg
-dependencies:
-  foo:
-    hosted: https://example.org/pub/
-''',
-              (pubspec) => pubspec.dependencies,
-              'The syntax `hosted: <hosted-url> requires a` minimum Dart SDK '
-                  'constraint of 2.15!');
```

Comment:
hmm, yeah, it's probably best that find a way to produce an error like this... Rather than interpreting the URL as a package name.

We could also print a warning.. But maybe it's enough to be able to detect the difference between a package name and a URL.